src/Physics/Evaluate_friction_law.f90

```diff
@@ -526,6 +536,109 @@ SUBROUTINE Linear_slip_weakening_TPV1617(TractionGP_XY,TractionGP_XZ,       & !
+  SUBROUTINE ImposedSlipRateOnDRBoundary(TractionGP_XY,TractionGP_XZ,       & ! OUT: traction
+                                   NorStressGP,XYStressGP,XZStressGP,         & ! IN: Godunov status
+                                   iFace,iSide,iElem,nBndGP,nTimeGP,          & ! IN: element ID and GP lengths
+                                   rho,rho_neig,w_speed,w_speed_neig,         & ! IN: background values
+                                   time,DeltaT,                               & ! IN: time
+                                   DISC,EQN,MESH,MPI,IO)
+    !-------------------------------------------------------------------------!
+    IMPLICIT NONE
+    !-------------------------------------------------------------------------!
+    TYPE(tEquations)               :: EQN
+    TYPE(tDiscretization), target  :: DISC
+    TYPE(tUnstructMesh)            :: MESH
+    TYPE(tMPI)                     :: MPI
+    TYPE(tInputOutput)             :: IO    
+    INTEGER     :: iBndGP,iTimeGP,nBndGP,nTimeGP
+    INTEGER     :: iFace,iSide,iElem
+    REAL        :: time
+    REAL        :: NorStressGP(nBndGP,nTimeGP)
+    REAL        :: XYStressGP(nBndGP,nTimeGP)
+    REAL        :: XZStressGP(nBndGP,nTimeGP)
+    REAL        :: TractionGP_XY(nBndGP,nTimeGP)
+    REAL        :: TractionGP_XZ(nBndGP,nTimeGP)
+    real        :: LocTracXY(nBndGP)
+    real        :: LocTracXZ(nBndGP)
+    real        :: tmpSlip(nBndGP)
+    real        :: Strength(nBndGP), ShTest(nBndGP)
+    real        :: LocSR(nBndGP)
+    REAL        :: rho,rho_neig,w_speed(:),w_speed_neig(:)
+    REAL        :: time_inc
+    REAL        :: Deltat(1:nTimeGP)
+    REAL        :: Tnuc, Gnuc, Gnucprev, dt, prevtime
+    real        :: tn, eta
+    !-------------------------------------------------------------------------!
+    INTENT(IN)    :: NorStressGP,XYStressGP,XZStressGP,iFace,iSide,iElem
+    INTENT(IN)    :: rho,rho_neig,w_speed,w_speed_neig,time,nBndGP,nTimeGP,DeltaT
+    INTENT(IN)    :: MESH,MPI,IO
+    INTENT(INOUT) :: EQN, DISC,TractionGP_XY,TractionGP_XZ
+    !-------------------------------------------------------------------------! 
+    Tnuc = DISC%DynRup%t_0
+    tmpSlip = 0.0D0
+    
+    eta = (w_speed(2)*rho*w_speed_neig(2)*rho_neig) / (w_speed(2)*rho + w_speed_neig(2)*rho_neig)
+    tn = time
+    
+    dt = sum(DeltaT(:))
+
+    do iTimeGP=1,nTimeGP
+      time_inc = DeltaT(iTimeGP)
+      prevtime = tn
+      tn=tn + time_inc
+
+      IF (time.LE.Tnuc) THEN
+         Gnucprev = EXP((prevtime-Tnuc)**2/(prevtime*(prevtime-2.0D0*Tnuc)))
+         Gnuc=EXP((tn-Tnuc)**2/(tn*(tn-2.0D0*Tnuc))) - Gnucprev
+         Gnuc=Gnuc/time_inc
+      ELSE
+         Gnuc=0.0D0
+      ENDIF
+
+      !EQN%NucleationStressInFaultCS (4 and 6) contains the slip in FaultCS
+      LocTracXY(:)  = XYStressGP(:,iTimeGP) + eta * EQN%NucleationStressInFaultCS(:,4,iFace)*Gnuc
+      LocTracXZ(:) =  XZStressGP(:,iTimeGP) + eta * EQN%NucleationStressInFaultCS(:,6,iFace)*Gnuc
```

Comment:
Should be "-" because otherwise you change the plus-side and minus-side convention.